parameter/command-processor/include/CommandParser.h

```diff
@@ -0,0 +1,222 @@
+
+    CommandReturn showProperties(const IRemoteCommand& remoteCommand, std::string& result);
+
+    CommandReturn getParameter(const IRemoteCommand& remoteCommand, std::string& result);
+
+    CommandReturn setParameter(const IRemoteCommand& remoteCommand, std::string& result);
+
+    CommandReturn getConfigurationParameter(const IRemoteCommand& remoteCommand,
+                                            std::string& result);
+
+    CommandReturn setConfigurationParameter(const IRemoteCommand& remoteCommand,
+                                            std::string& result);
+
+    CommandReturn listBelongingDomains(const IRemoteCommand& remoteCommand, std::string& result);
+
+    CommandReturn listAssociatedDomains(const IRemoteCommand& remoteCommand,
+                                        std::string& result);
+
+    CommandReturn showMapping(const IRemoteCommand& remoteCommand, std::string& result);
+
+    CommandReturn listAssociatedElements(const IRemoteCommand& remoteCommand, std::string& result);
+
+    CommandReturn listConflictingElements(const IRemoteCommand& remoteCommand, std::string& result);
+
+    CommandReturn listRogueElements(const IRemoteCommand& remoteCommand, std::string& result);
+
+    CommandReturn exportConfigurableDomainsToXML(const IRemoteCommand& remoteCommand,
+                                                 std::string& result);
+
+    CommandReturn importConfigurableDomainsFromXML(const IRemoteCommand& remoteCommand,
+                                                   std::string& result);
+
+    CommandReturn exportConfigurableDomainsWithSettingsToXML(const IRemoteCommand& remoteCommand,
+                                                             std::string& result);
+
+    CommandReturn importConfigurableDomainsWithSettingsFromXML(const IRemoteCommand& remoteCommand,
+                                                               std::string& result);
+
+    CommandReturn importConfigurableDomainWithSettingsFromXML(const IRemoteCommand& remoteCommand,
+                                                              std::string& result);
+
+    CommandReturn exportSettings(const IRemoteCommand& remoteCommand, std::string& result);
+
+    CommandReturn importSettings(const IRemoteCommand& remoteCommand, std::string& result);
+
+    CommandReturn getConfigurableDomainsWithSettingsXML(const IRemoteCommand& remoteCommand,
+                                                        std::string& result);
+
+    CommandReturn getConfigurableDomainWithSettingsXML(const IRemoteCommand& remoteCommand,
+                                                       std::string& result);
+
+    CommandReturn setConfigurableDomainsWithSettingsXML(const IRemoteCommand& remoteCommand,
+                                                        std::string& result);
+
+    CommandReturn getSystemClassXML(const IRemoteCommand& remoteCommand, std::string& result);
```

Comment:
those handler are so ugly, maybe one day it will be factorized with either command line options or c++ template.